spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ControllerHelperServiceUtils.java

```diff
@@ -1,19 +1,143 @@
 package uk.ac.bbsrc.tgac.miso.spring;
 
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringEmptyOrNull;
+
 import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 import javax.servlet.http.HttpSession;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sourceforge.fluxion.ajax.util.JSONUtils;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Barcodable;
+import uk.ac.bbsrc.tgac.miso.core.data.Printer;
+import uk.ac.bbsrc.tgac.miso.service.PrinterService;
+
 /**
  * For when you need web utils, since this module can't access `miso-web`.
  *
  */
 public class ControllerHelperServiceUtils {
+  private static final Logger log = LoggerFactory.getLogger(ControllerHelperServiceUtils.class);
+
+  public interface BarcodePrintAssister<T extends Barcodable> {
+
+    /**
+     * Get all the items associated with a project or throw if this is not reasonable.
+     * 
+     * @param projectId the project in question
+     * @return a collection of assoicated items
+     * @throws IOException
+     */
+    public Iterable<T> fetchAll(long projectId) throws IOException;
+
+    /**
+     * Load an item from the database by ID.
+     * 
+     * @param id
+     * @return
+     * @throws IOException
+     */
+    public T fetch(long id) throws IOException;
+
+    /**
+     * Write an item to the database having updated the identification barcode.
+     * 
+     * @param item
+     * @throws IOException
+     */
+    public void store(T item) throws IOException;
+
+    /**
+     * The JSON property in the request for the list of objects to fetch.
+     */
+    public String groupName();
+
+    /**
+     * The JSON property containing the ID of the object to be fetched.
+     */
+    public String idName();
```

Comment:
getIdName and getGroupName would be preferable method names as verbs are typically expected in Java, and both "group" and "id" could be used as verbs. The Javadocs help clarify though, so either way